improver/spotdata/apply_lapse_rate.py

```diff
@@ -106,27 +110,97 @@ def process(
             A copy of the input spot_data_cube with the data modified by
             the lapse rates to give a better representation of the site's
             temperatures.
-        """
-        # Check the cubes are compatible.
-        check_grid_match([neighbour_cube, spot_data_cube, gridded_lapse_rate_cube])
 
-        # Extract the lapse rates that correspond to the spot sites.
-        spot_lapse_rate = SpotExtraction(
-            neighbour_selection_method=self.neighbour_selection_method
-        )(neighbour_cube, gridded_lapse_rate_cube)
-
-        # Extract vertical displacements between the model orography and sites.
-        method_constraint = iris.Constraint(
-            neighbour_selection_method_name=self.neighbour_selection_method
-        )
-        data_constraint = iris.Constraint(grid_attributes_key="vertical_displacement")
-        vertical_displacement = neighbour_cube.extract(
-            method_constraint & data_constraint
-        )
-
-        # Apply lapse rate adjustment to the temperature at each site.
-        new_temperatures = (
-            spot_data_cube.data + (spot_lapse_rate.data * vertical_displacement.data)
-        ).astype(np.float32)
-        new_spot_cube = spot_data_cube.copy(data=new_temperatures)
-        return new_spot_cube
+        Raises:
+            ValueError:
+                If the lapse rate cube was provided but the diagnostic being
+                processed is not air temperature.
+            ValueError:
+                If the lapse rate cube provided does not have the name
+                "air_temperature_lapse_rate"
+            ValueError:
+                If the lapse rate cube does not contain a single valued height
+                coordinate.
+
+        Warns:
+            warning:
+                If a lapse rate cube was provided, but the height of the
+                temperature does not match that of the data used.
+        """
+        if not is_probability(spot_data_cube):
```

Comment:
Do the `if is_probability...` first, then you won't need to put all this into a conditional block at all. Black will then hopefully unsplit some of the lines again.

```suggestion
        if is_probability(spot_data_cube):
            raise ValueError(etc...)
    
        # Check that we are dealing with temperature data.
        ....
```

Reply:
Done